sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -22,17 +27,246 @@ class TableCheckpointStore:
+        except ResourceNotFoundError:
+            self._create_entity_checkpoint(checkpoint)
+
+    def _upload_ownership(self, ownership):
+        try:
+            theentity = self._modify_entity_ownership(ownership)
+            entity = self.table_client.update_entity(mode=UpdateMode.REPLACE, entity=theentity,
+            etag=ownership['etag'], match_condition=MatchConditions.IfNotModified)
+            ownership['etag'] = entity['etag']
+            ownership['last_modified_time'] = entity['date']
+        except (ResourceNotFoundError, ValueError):
+            try:
+                entity = self._create_entity_ownership(ownership)
+                ownership['etag'] = entity['etag']
+                ownership['last_modified_time'] = entity['date']
+            except ResourceExistsError:
+                raise'your etag does not match the last time this ownership was modified.'
+        except ResourceModifiedError:
+            raise'your etag does not match the last time this ownership was modified.'
 
-    def list_checkpoints(self, namespace, eventhub, consumergroup, **kwargs):
-        pass
 
-    def update_checkpoint(self, checkpoint, **kwargs):
-        pass
+    def _claim_one_partition(self, ownership):
+        self._upload_ownership(ownership)
+        return ownership
 
-    def claim_ownership(self, ownershiplist, **kwargs):
-        pass
+    def claim_ownership(self, ownershiplist):
+        # type: (Iterable[Dict[str, Any]], Any) -> Iterable[Dict[str, Any]]
+        """Tries to claim ownership for a list of specified partitions.
+        :param Iterable[Dict[str,Any]] ownership_list: Iterable of dictionaries containing all the ownerships to claim.
+        :rtype: Iterable[Dict[str,Any]], Iterable of dictionaries containing partition ownership information:
+                - `fully_qualified_namespace` (str): The fully qualified namespace that the Event Hub belongs to.
+                  The format is like "<namespace>.servicebus.windows.net".
+                - `eventhub_name` (str): The name of the specific Event Hub the checkpoint is associated with,
+                  relative to the Event Hubs namespace that contains it.
+                - `consumer_group` (str): The name of the consumer group the ownership are associated with.
+                - `partition_id` (str): The partition ID which the checkpoint is created for.
+                - `owner_id` (str): A UUID representing the owner attempting to claim this partition.
+                - `last_modified_time` (UTC datetime.datetime): The last time this ownership was claimed.
+                - `etag` (str): The Etag value for the last time this ownership was modified. Optional depending
+                  on storage implementation.
+        """
+        newlist = []
+        for x in ownershiplist:
+            newlist.append(self._claim_one_partition(x))
+        return newlist
```

Comment:
It looks like `claim_ownership` mutates each ownership that's in the `ownershiplist`. It actually looks like the storage-blob implementation does this too, but I wouldn't have expected that.

@swathipil Am I just totally wrong about python and it is pass by value instead of pass by reference? Or is this indeed mutating each ownership the user passes in (albeit appending them to a new list).

Reply:
yep, you're correct - python is pass by reference and it is mutating the original ownership that is passed in. I'm not sure why it was implemented this way. Will check with Adam to see if he knows.

Reply:
so it looks like this is a bug and we'll be fixing it, thanks for catching this @chradek !

Reply:
@Jg1255 
So currently this method is mutating each ownership that the user is passing in, rather than returning a list of new ownership. Can you make sure that the ownership passed in isn't mutated?

Reply:
comment above regarding how to deal with this @Jg1255